cmd/syft/cli/options/source.go

```diff
@@ -12,16 +12,40 @@ type sourceConfig struct {
 	Version  string      `json:"version" yaml:"version" mapstructure:"version"`
 	BasePath string      `yaml:"base-path" json:"base-path" mapstructure:"base-path"` // specify base path for all file paths
 	File     fileSource  `json:"file" yaml:"file" mapstructure:"file"`
+	Image    imageSource `json:"image" yaml:"image" mapstructure:"image"`
 }
 
 type fileSource struct {
 	Digests []string `json:"digests" yaml:"digests" mapstructure:"digests"`
 }
 
-func defaultSourceCfg() sourceCfg {
-	return sourceCfg{
+type imageSource struct {
+	DefaultPullSource string `json:"default-pull-source" yaml:"default-pull-source" mapstructure:"default-pull-source"`
+}
+
+func defaultSourceCfg() sourceConfig {
+	return sourceConfig{
 		File: fileSource{
 			Digests: []string{"sha256"},
 		},
+		Image: imageSource{
+			DefaultPullSource: "",
+		},
+	}
+}
+
+func (c imageSource) PostLoad() error {
+	return checkDefaultSourceValues(c.DefaultPullSource)
+}
+
+var validDefaultSourceValues = []string{"registry", "docker", "podman", ""}
```

Comment:
What does `""` mean as a default source?

Reply:
this is the default value and defers to the underlying lib to control this behavior